djconnectwise/sync.py

```diff
@@ -205,8 +205,13 @@ def get_single(self, *args, **kwargs):
         raise NotImplementedError
 
     def fetch_sync_by_id(self, instance_id):
+        instance = None
         api_instance = self.get_single(instance_id)
-        instance, created = self.update_or_create_instance(api_instance)
+        try:
+            instance, created = self.update_or_create_instance(api_instance)
```

Comment:
This may not be the best place to catch this exception. Depending on when we encounter this exception, we may want to do different things. 

This is similar to the work we just did in dja, where we let the synchronizers fail and raise their exceptions, and the methods calling them decide what to do when they encounter those exceptions. If we catch the exception here then anything that calls `fetch_sync_by_id` MUST handle the exception this way. A better place would be to catch it where fetch_sync_by_id is being called from, since it is already wrapped in a try/catch for `ConnectWiseAPIError`, it wouldn't be much more to add `InvalidObjectException`.

Reply:
Yeah I see. That makes sense. I saw that in `persist_page` there was a try except for `InvalidObjectException` and figured it'd make sense to do the same thing in `fetch_sync_by_id`. 

Not catching the exception in `fetch_sync_by_id` does seem more flexible. I'll make the changes. Closed a little prematurely but this branch is no longer needed.